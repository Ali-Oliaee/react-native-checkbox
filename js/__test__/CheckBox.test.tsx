import * as React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '../CheckBox.android';

describe('<Checkbox />', () => {
  it('renders enabled Checkbox', () => {
    const tree = renderer.create(<Checkbox />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled Checkbox', () => {
    const tree = renderer.create(<Checkbox disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders Checkbox with value', () => {
    const tree = renderer.create(<Checkbox value />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
