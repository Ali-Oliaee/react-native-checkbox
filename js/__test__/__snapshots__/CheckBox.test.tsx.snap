// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders Checkbox with value 1`] = `
<AndroidCheckBox
  enabled={true}
  on={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 32,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Checkbox /> renders disabled Checkbox 1`] = `
<AndroidCheckBox
  enabled={false}
  on={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 32,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Checkbox /> renders enabled Checkbox 1`] = `
<AndroidCheckBox
  enabled={true}
  on={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 32,
      },
      undefined,
    ]
  }
/>
`;
